src/Pact/Eval.hs

```diff
@@ -305,27 +305,38 @@ toPersistDirect' t = case toPersistDirect t of
 
 
 evalUse :: Use -> Eval e ()
-evalUse (Use mn h i) = do
+evalUse (Use mn h mis i) = do
   mm <- resolveModule i mn
   case mm of
     Nothing -> evalError i $ "Module " <> pretty mn <> " not found"
     Just md -> do
-      case view mdModule md of
+      case _mdModule md of
         MDModule Module{..} ->
           case h of
             Nothing -> return ()
             Just mh | mh == _mHash -> return ()
                     | otherwise -> evalError i $ "Module " <>
                         pretty mn <> " does not match specified hash: " <>
                         pretty mh <> ", " <> pretty _mHash
-        MDInterface Interface{..} ->
+        MDInterface i' ->
           case h of
             Nothing -> return ()
-            Just _ -> evalError i $
-              "Interfaces should not have associated hashes: " <>
-              pretty _interfaceName
+            Just _ -> evalError i
+              $ "Interfaces should not have associated hashes: "
+              <> pretty (_interfaceName i')
 
-      installModule False md
+      validateImports i mis (_mdRefMap md)
+      installModule False mis md
+
+validateImports :: Info -> Maybe (V.Vector Text) -> HM.HashMap Text Ref -> Eval e ()
+validateImports i mis rs = maybe (return ()) (traverse_ go) mis
+  where
+    go imp = case HM.lookup imp rs of
+      Nothing -> evalError i $ "imported name not found: " <> pretty imp
+      Just (Ref (TDef d _)) -> case _dDefType d of
+        Defcap -> evalError i $ "cannot import capabilities: " <> pretty imp
+        _ -> return ()
+      Just _ -> return ()
```

Comment:
We could probably restrict this further so it fails on load before runtime. Thoughts @slpopejoy?